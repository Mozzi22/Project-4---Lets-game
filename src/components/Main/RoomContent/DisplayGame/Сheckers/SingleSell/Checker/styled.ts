import styled from 'styled-components';

export const StChecker = styled.div<{bgColors: string}>`
    border: 2px solid black;
    display: flex;
    height: 40px;
    width: 40px;
    border-radius: 50%;
    align-items: center;
    justify-content: center;
    box-shadow: 0 0 2px 2px black;
    background-color: ${({ bgColors = 'green' }) => bgColors};
    &::after {
        border: 2px solid black;
        background-color: ${({ bgColors = 'green' }) => bgColors};

    }
`;
