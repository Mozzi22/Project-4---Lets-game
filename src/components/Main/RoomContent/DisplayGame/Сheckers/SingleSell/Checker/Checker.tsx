import React from 'react';
import { TCurrentChecker } from 'src/Store/games/types';
import { StChecker } from './styled'
interface IProps {
    color: string | null;
    position: number;
    isChecked: boolean | null;
    setCurrentChecker: (arg: TCurrentChecker) => void;
}

const Checker: React.FC<IProps> = ({ setCurrentChecker, color, position, isChecked  }) => {
    const onCheckerClick = () => {
        setCurrentChecker({ currentChecker: { color, position, isChecked } })
    };
    return (
        <StChecker
            bgColors={color === 'black'
                ? 'black'
                : 'white'}
            onClick={onCheckerClick}
        />
    );
};

export default Checker;
