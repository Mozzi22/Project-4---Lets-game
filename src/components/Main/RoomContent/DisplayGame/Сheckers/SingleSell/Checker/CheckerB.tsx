import React from 'react';
import { TCheckers } from 'src/components/Main/RoomContent/DisplayGame/Ð¡heckers/types';
import { StChecker } from './styled'

const CheckerWhite: React.FC<TCheckers> = ({ bgColors, id }) => {
    return (
        <StChecker
            id={id}
            bgColors= {bgColors}
        />
    );
};

export default CheckerWhite;
