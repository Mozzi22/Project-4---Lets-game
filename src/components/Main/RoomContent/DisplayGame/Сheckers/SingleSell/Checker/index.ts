// import { connect } from 'react-redux';
// import { getCurrentChecker } from 'src/Store/games/selectors';
// import { setCurrentChecker } from 'src/Store/games/actions';
// import { ApplicationState } from 'src/Store/types';
import Checker from './Checker';
// import { Dispatch } from 'redux';
// import { TCurrentChecker } from 'src/Store/games/types';

// const mapStateToProps = (state: ApplicationState) => ({
//   currentChecker: getCurrentChecker(state),
// });
// const mapDispatchToProps = (dispatch: Dispatch) => ({
//   setCurrentChecker: (payload) => dispatch(setCurrentChecker(payload)),
// })

export default Checker;
