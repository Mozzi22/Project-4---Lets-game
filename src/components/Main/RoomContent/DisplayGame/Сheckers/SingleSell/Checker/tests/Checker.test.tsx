import React from 'react';
import { shallowSmart, mountSmart } from 'src/helpers/tests/testHelper';
import Checker from '../Checker';

describe('Checkers', () => {
    const props = {
        colors: 'black',
        setCurrentChecker: jest.fn(),
        position: 6,
        isChecked: true,
    }
    it('Should match snapshot', () => {
        const component = shallowSmart(<Checker {...props}/>);
        expect(component.html()).toMatchSnapshot();
    });
    it('should render StCheckers', () => {
        const component = mountSmart(<Checker {...props}/>);
        expect(component.find('styled__StChecker')).toHaveLength(1);
    });
      it('should click on the button', () => {
          const component = mountSmart(<Checker {...props} />);
          component.find('styled__StChecker').props().onClick();
          expect(props.setCurrentChecker).toHaveBeenCalled();
    });
});