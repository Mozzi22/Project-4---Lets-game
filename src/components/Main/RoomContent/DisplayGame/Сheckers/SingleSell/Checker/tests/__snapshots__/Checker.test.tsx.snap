// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkers Should match snapshot 1`] = `"<div class=\\"styled__StChecker-sc-3d1dux-0 bUcYpr\\"></div>"`;
